### PR TITLE
Update modeling_tf_bert.py

### PR DESCRIPTION
**Pull request by BruceJust**
# What does this PR do?
change the attention_mask buid code in modeling_tf_bert.py to support customer attention, such as unilm_mask
former：           
 extended_attention_mask = tf.reshape(
                inputs["attention_mask"], (attention_mask_shape[0], 1, 1, attention_mask_shape[1])
            )
new：
  extended_attention_mask = tf.reshape(
                inputs["attention_mask"], (attention_mask_shape[0], 1, -1, attention_mask_shape[1])
            )

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sgugger @LysandreJik

### CODE REVIEW

**Comment by BruceJust** [comment]
this change is very necessary to allowed researcher to build a customized attention, and i foud it has been done in pytorch version.

**Comment by BruceJust** [comment]
this change is very necessary to allowed researcher to build a customized attention, and i foud it has been done in pytorch version.

File: src/transformers/models/bert/modeling_tf_bert.py [hunk 1]
```diff
@@ -821,7 +821,7 @@ def call(
             )
         else:
             extended_attention_mask = tf.reshape(
-                inputs["attention_mask"], (attention_mask_shape[0], 1, 1, attention_mask_shape[1])
+                inputs["attention_mask"], (attention_mask_shape[0], 1, -1, attention_mask_shape[1])
```
Comment (sgugger) [inline src/transformers/models/bert/modeling_tf_bert.py:824]: I don't think this is the right fix. When the attention mask is going to be 3d, we should use `attention_mask_shape[2]` for the last dimension. The generic fix should thus be:
```suggestion
                inputs["attention_mask"], (attention_mask_shape[0], 1, -1, attention_mask_shape[-1])
```

**Review by sgugger — COMMENTED** [review]
Thanks a lot for your PR, I left a comment as I don't think this really fixes the problem yet. Then this should perhaps be used in more models than just this one?

cc @LysandreJik @Rocketknight1

File: src/transformers/models/bert/modeling_tf_bert.py (hunk 1 above)
Comment (BruceJust) [reply src/transformers/models/bert/modeling_tf_bert.py:824]: thanks for your comment, and you are right, this is a lazy way, we can make it the same way as pytorch version:
line 262 in modeling_utils
        if attention_mask.dim() == 3:
            extended_attention_mask = attention_mask[:, None, :, :]
        elif attention_mask.dim() == 2:
            .......
how do you think?

Comment (BruceJust) [reply src/transformers/models/bert/modeling_tf_bert.py:824]: and yes, it should be a common function for all transformer models in tf version
